frontend/control-center/src/pages/Connectors/ConnectorConfig/index.tsx

```diff
@@ -153,114 +154,18 @@ const ConnectorConfig = (props: ConnectorConfigProps) => {
-        appCredentials,
-        suggestionConfidenceLevel,
-        replyConfidenceLevel,
-        processorWaitingTime,
-        processorCheckPeriod,
-        defaultLanguage,
-      ] = args;
-
-      payload = {
-        components: [
-          {
-            name: connectorInfo && removePrefix(connectorInfo.name),
-            enabled: true,
-            data: {
-              projectId: projectId,
-              dialogflowCredentials: appCredentials,
-              suggestionConfidenceLevel: suggestionConfidenceLevel,
-              replyConfidenceLevel: replyConfidenceLevel,
-              connectorStoreMessagesProcessorMaxWaitMillis: processorWaitingTime,
-              connectorStoreMessagesProcessorCheckPeriodMillis: processorCheckPeriod,
-              connectorDefaultLanguage: defaultLanguage,
-            },
-          },
-        ],
-      };
-    }
-
-    if (connector === Source.zendesk) {
-      const [domain, token, username] = args;
-
-      payload = {
-        components: [
-          {
-            name: connectorInfo && removePrefix(connectorInfo.name),
-            enabled: true,
-            data: {
-              domain: domain,
-              token: token,
-              username: username,
-            },
-          },
-        ],
-      };
-    }
-
-    if (connector === Source.salesforce) {
-      const [url, username, password, securityToken] = args;
-
-      payload = {
-        components: [
-          {
-            name: connectorInfo && removePrefix(connectorInfo.name),
-            enabled: true,
-            data: {
-              url: url,
-              username: username,
-              password: password,
-              securityToken: securityToken,
-            },
-          },
-        ],
-      };
-    }
-
-    if (connector === Source.rasa) {
-      const [webhookUrl, apiHost, token] = args;
-
-      payload = {
-        components: [
-          {
-            name: connectorInfo && removePrefix(connectorInfo.name),
-            enabled: true,
-            data: {
-              webhookUrl: webhookUrl,
-              apiHost: apiHost,
-              token: token,
-            },
-          },
-        ],
-      };
-    }
-
-    if (connector === Source.whatsapp) {
-      const [appId, appSecret, phoneNumber, name, avatarUrl] = args;
-
-      payload = {
-        components: [
-          {
-            name: connectorInfo && removePrefix(connectorInfo.name),
-            enabled: true,
-            data: {
-              appId: appId,
-              appSecret: appSecret,
-              phoneNumber: phoneNumber,
-              name: name,
-              avatarUrl: avatarUrl,
-            },
-          },
-        ],
-      };
-    }
+    const payload: UpdateComponentConfigurationRequestPayload = {
+      components: [
+        {
+          name: connectorInfo && removePrefix(connectorInfo.name),
+          enabled: true,
+          data: configurationValues,
+        },
+      ],
+    };
```

Comment:
we do not need to hardcode the configuration of each component: we can just prepare the payload dynamically